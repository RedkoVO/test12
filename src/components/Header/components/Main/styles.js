import BitcoinLogo from '../../../../assets/images/svg/bitcoin.svg'
import DownLogo from '../../../../assets/images/svg/down.svg'

export default () => ({
  root: {
    display: 'flex',
    flexDirection: 'column',
    '-ms-align-items': 'center',
    alignItems: 'center',
    padding: '30px 25px 0 25px',
    width: '100%',
    boxSizing: 'border-box'
  },
  user: {
    display: 'flex',
    flexDirection: 'row',
    justifyContent: 'flex-end',
    width: '100%'
  },
  userBlock: {
    display: 'flex',
    '-ms-align-items': 'center',
    alignItems: 'center'
  },
  userHello: {
    fontSize: '18px',
    margin: '0 0 20px 0',
  },
  userVerification: {
    fontSize: '10px',
    fontWeight: '500'
  },
  wallet: {
    display: 'flex',
    flexDirection: 'column',
    '-ms-align-items': 'stretch',
    alignItems: 'stretch',
    width: 'initial'
  },
  walletTitle: {
    textAlign: 'center',
    fontSize: '39px',
    fontWeight: 400,

    '& p': {
      margin: '30px 0 0 0'
    }
  },
  walletSum: {
    position: 'relative',
    padding: 0,
    paddingLeft: '0',
    boxSizing: 'border-box',

    '&::before': {
      // position: 'absolute',
      // content: `""`,
      // width: '45px',
      // height: '45px',
      // bottom: '15px',
      // left: '50px',
      // backgroundImage: `url(${BitcoinLogo})`,
      // backgroundRepeat: 'no-repeat',
      // backgroundSize: '45px',
      // backgroundPosition: 'center'
    },

    '&::after': {
      position: 'absolute',
      content: `""`,
      width: '30px',
      height: '30px',
      bottom: '25px',
      right: '-35px',
      backgroundImage: `url(${DownLogo})`,
      backgroundRepeat: 'no-repeat',
      backgroundSize: '30px',
      backgroundPosition: 'center'
    },

    '& a': {
      color: '#5e7cff',
      fontSize: '82px',
      fontWeight: 700,
      margin: '15px 0 0 0',
      padding: 0,
      lineHeight: '82px',
      textDecoration: 'none'
    }
  },
  walletNav: {
    display: 'flex',
    flexDirection: 'row',
    listStyleType: 'none',
    margin: 0,
    width: '100%',
    justifyContent: 'center'
  },
  walletNavItem: {
    textDecoration: 'none',
    color: '#f0f0f0',
    backgroundColor: '#5e7cff',
    fontWeight: 700,
    fontSize: '10px',
    padding: '5px 15px',
    borderRadius: '10px',
    margin: '0 5px'
  }
})