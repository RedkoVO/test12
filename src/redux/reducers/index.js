import { combineReducers } from 'redux'
import { reducer as form } from 'redux-form'
import auth from './auth'
import balance from './balance'

export default combineReducers({
  form,
  auth,
  balance
})