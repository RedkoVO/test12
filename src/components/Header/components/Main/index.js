import React from 'react'
import PropTypes from 'prop-types'
import withStyles from '@material-ui/core/styles/withStyles'

import Action from './components/Action'

import Avatar from '../../../../assets/images/user_photo.png'

import styles from './styles'

const Main = ({ classes, onSubmit, balance, isDisabledButton }) => (
  <div className={classes.root}>
    <div className={classes.demoTitle}>
      DEMO VERSION
    </div>
    <div className={classes.user}>
      <div className={classes.userBlock}>
        <p className={classes.userHello}> Hello, Ben A <br />
          <span className={classes.userVerification}>KYC VERIFICATED</span>
        </p>
        <div className={classes.userPhoto}>
          <img src={Avatar} alt="" />
        </div>
      </div>
    </div>

    <div className={classes.wallet}>
      <div className={classes.walletTitle}>
        <p>My dcb wallet:</p>
      </div>
      <div className={classes.walletSum}>
        <a href="/">{balance}</a>
      </div>
      <ul className={classes.walletNav}>
        <li>
          <a href="/" className={classes.walletNavItem}>Send Money</a>
        </li>
        <li>
          <a href="/" className={classes.walletNavItem}>Tranfer Money</a>
        </li>
        <li>
          <a href="/" className={classes.walletNavItem}>Invoice</a>
        </li>
        <li>
          <a href="/" className={classes.walletNavItem}>Cashout</a>
        </li>
      </ul>
    </div>

    <Action onSubmit={onSubmit} isDisabledButton={isDisabledButton} />
  </div>
)

Main.propTypes = {
  classes: PropTypes.object,
  onSubmit: PropTypes.func,
  balance: PropTypes.string,
  isDisabledButton: PropTypes.bool
}

export default withStyles(styles)(Main)