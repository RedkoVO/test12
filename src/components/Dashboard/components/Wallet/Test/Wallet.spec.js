import * as React from 'react'
import renderer from 'react-test-renderer'
import { Provider } from 'react-redux'
import { reduxForm } from 'redux-form'
import { createStore } from 'redux'

import Main from '../'

const allBalanceResultMock = [
  {
    balance: '0',
    currency: 'DCB',
    lastBlock:
      '0000000000000000000000000000000000000000000000000000000000000000'
  }
]

const Wrapper = reduxForm({ form: 'test' })(({ children }) => children)
const store = createStore(a => a, {})

it('renders correctly', () => {
  const tree = renderer
    .create(
      <Provider store={store}>
        <Wrapper>
          <Main
            onSubmit={() => {}}
            handleChangeBalance={() => {}}
            allBalanceResult={allBalanceResultMock}
            curencySelectValue={'USD'}
            isDisabledButton={false}
          />
        </Wrapper>
      </Provider>
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})
