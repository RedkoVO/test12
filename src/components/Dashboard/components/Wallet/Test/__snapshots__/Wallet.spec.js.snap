// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="Wallet-root-1"
>
  <div
    className="Wallet-wallet-2"
  >
    <div
      className="Wallet-walletTitle-3"
    >
      <p>
        My Address:
      </p>
    </div>
    <div
      className="Wallet-walletAddress-4"
    />
    <div
      className="MuiInputBase-root-32 MuiInput-root-19 MuiInput-underline-23 Wallet-selectCurency-5"
      onClick={[Function]}
    >
      <div
        className="MuiSelect-root-12"
      >
        <div
          aria-haspopup="true"
          aria-pressed="false"
          className="Wallet-selectField-7"
          id="select-curency"
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          role="button"
          tabIndex={0}
        >
          <span
            dangerouslySetInnerHTML={
              Object {
                "__html": "&#8203;",
              }
            }
          />
        </div>
        <input
          id="curencyDropdownWallet"
          name="curency"
          type="hidden"
          value="DCB"
        />
        <i
          className="Wallet-curencyIcon-8"
        />
      </div>
    </div>
    <ul
      className="Wallet-walletNav-10"
    >
      <li>
        <div
          className="Wallet-walletNavItem-11"
          onClick={[Function]}
        >
          DCB
        </div>
      </li>
      <li>
        <div
          className="Wallet-walletNavItem-11"
          onClick={[Function]}
        >
          USD
        </div>
      </li>
      <li>
        <div
          className="Wallet-walletNavItem-11"
          onClick={[Function]}
        >
          EUR
        </div>
      </li>
      <li>
        <div
          className="Wallet-walletNavItem-11"
          onClick={[Function]}
        >
          CPT
        </div>
      </li>
    </ul>
  </div>
  <div
    className="Action-root-53"
  >
    <div
      className="Action-actionsSend-58"
    >
      <form
        className="Action-form-56"
        onSubmit={[Function]}
      >
        <span
          className="Action-sendTitle-59"
        >
          Send now
        </span>
        <div
          className="Action-wrSendField-60"
        >
          <label
            htmlFor="addressKey"
          >
            Address:
          </label>
          <input
            className="Action-sendAddressField-61"
            onBlur={[Function]}
            onChange={[Function]}
            onDragStart={[Function]}
            onDrop={[Function]}
            onFocus={[Function]}
            placeholder="Put address you want to send the money"
            type="text"
            value=""
          />
        </div>
        <div
          className="Action-wrSendField-60"
        >
          <label
            htmlFor="amount"
          >
            Amount:
          </label>
          <input
            className="Action-sendAmmountField-62"
            onBlur={[Function]}
            onChange={[Function]}
            onDragStart={[Function]}
            onDrop={[Function]}
            onFocus={[Function]}
            placeholder="Set ammount"
            type="text"
            value=""
          />
        </div>
        <button
          className="Action-sendBtn-64"
          type="submit"
        >
          Send money
        </button>
      </form>
    </div>
    <div
      className="Action-actionsInvoice-66"
    >
      <span
        className="Action-invoiceTitle-67"
      >
        Invoice
      </span>
      <div
        className="Action-wrSendField-60"
      >
        <label
          htmlFor="invoice"
        >
          Email:
        </label>
        <input
          className="Action-invoiceAmmountField-63"
          name="invoice"
          placeholder="Set ammount you need to receive"
          type="text"
        />
      </div>
      <button
        className="Action-sendBtn-64"
        type="submit"
      >
        Send Invoice
      </button>
    </div>
  </div>
</div>
`;
